pywatts/core/base_step.py

```diff
@@ -121,19 +121,21 @@ def further_elements(self, counter: pd.Timestamp) -> bool:
     def _pack_data(self, start, end, buffer_element=None, return_all=False):
         # Provide requested data
         time_index = _get_time_indexes(self.buffer)
-        if end and start and end > start:
+        if start:
             index = list(self.buffer.values())[0].indexes[time_index[0]]
-            start = max(index[0], start.to_numpy())
+            start = start.to_numpy()
+            # If end is not set, all values should be considered. Thus we add a small timedelta to the last index entry.
+            end = end.to_numpy() if end is not None else index[-1].to_numpy() + pd.Timedelta(milliseconds=1)
```

Comment:
If the sample size is in the area of milliseconds, this would be not a small timedelta. Therefore, would prefer to add a small fraction of the sample size instead of a millisecond.

Reply:
add .to_numpy() also to the pd.Timedelta, otherwise the types differ between the if and else

Reply:
I change it to nanoseconds. I suppose, it is the shortest available timeldelta in pandas.